set.mm

```diff
@@ -520296,6 +520699,33 @@ such that every nonzero prime ideal contains a prime element (this is a
       cidom isufd cdomn isidom abvn0b baib anbi2d anbi1d anass bitr2di ) BIJBKJ
       ZBLMZNOZCPAQNOCDERRUAUBZSSZBUCJZBUEJZURSZUPABCDEUPUDZFGHUFUTVBUOUQSZURSUS
       UTVAVDURVAUOBUGJZSUTVDBUHUTVEUQUOVEUTUQUPBVCUIUJUKTULUOUQURUMUNT $.
+
+    $d .0. j $.  $d I j $.  $d J j p $.  $d P j p $.  $d R j $.  $d j ph $.
+    ufdprmidl.1 $e |- ( ph -> R e. NzRing ) $.
+    ufdprmidl.2 $e |- ( ph -> R e. UFD ) $.
+    ufdprmidl.3 $e |- ( ph -> J e. I ) $.
+    ufdprmidl.4 $e |- ( ph -> J =/= { .0. } ) $.
+    $( In a unique factorization domains ` R ` , a nonzero prime ideal ` J `
+       contains a prime element ` p ` .  (Contributed by Thierry Arnoux,
+       3-Jun-2025.) $)
+    ufdprmidl $p |- ( ph -> E. p e. P p e. J ) $=
+      ( vj cv cin c0 wne wcel wrex csn cdif wa ineq1 neeq1d adantl incom neeq1i
+      wceq wb inn0 bitr3i bitrdi eldifsnd cnzr cufd wral cidom simplbda syl2anc
+      isufd2 rspcdv2 ) AOPZBQZRSZGPETGBUAZOEDFUBZUBUCZAVDEUJZUDVFEBQZRSZVGVJVFV
+      LUKAVJVEVKRVDEBUEUFUGVLBEQZRSVGVMVKRBEUHUIGBEULUMUNAEDVHMNUOACUPTZCUQTZVF
+      OVIURZKLVNVOCUSTVPBCODFHIJVBUTVAVC $.
+  $}
+
+  ${
+    $d R i $.
+    ufdidom.1 $e |- ( ph -> R e. NzRing ) $.
+    ufdidom.2 $e |- ( ph -> R e. UFD ) $.
+    $( A nonzero UFD is an integral domain.  (Contributed by Thierry Arnoux,
```

Comment:
Maybe it's better to expand "UFD" into "unique factorization domain" here, since you tend to use the non-abbreviated term in other theorems.